doc/api/esm.md

```diff
@@ -175,88 +175,75 @@ unspecified.
 
 ## Packages
 
-### Package Entry Points
+### Package Exports
 
 There are two fields that can define entry points for a package: `"main"` and
 `"exports"`. The `"main"` field is supported in all versions of Node.js, but its
 capabilities are limited: it only defines the main entry point of the package.
-The `"exports"` field, part of [Package Exports][], provides an alternative to
-`"main"` where the package main entry point can be defined while also
-encapsulating the package, preventing any other entry points besides those
-defined in `"exports"`. If package entry points are defined in both `"main"` and
-`"exports"`, the latter takes precedence in versions of Node.js that support
-`"exports"`. [Conditional Exports][] can also be used within `"exports"` to
-define different package entry points per environment.
 
-#### `package.json` `"main"`
+The `"exports"` field provides an alternative to `"main"` where the package
+main entry point can be defined while also encapsulating the package, preventing
+any other entry points besides those defined in `"exports"`. If package entry
+points are defined in both `"main"` and `"exports"`, the latter takes precedence
+in versions of Node.js that support `"exports"`. [Conditional Exports][] can
+also be used within `"exports"` to define different package entry points per
+environment.
 
-The `package.json` `"main"` field defines the entry point for a package,
-whether the package is included into CommonJS via `require` or into an ES
-module via `import`.
```

Comment:
So the previous text was organized as, "there are several entry points to a package: `"main"`, and `"exports"`. They function this way . . .". I think it's important we still document `"main"` and that it works for `import`/`import()`; it's not just ignored by ESM and used only by `require`. Arguably that's covered under the new section "Package Exports Main" but I could easily see people missing that, assuming that that section is referring only to how to define the main entry point within `"exports"`.

You could also consider the various paths/subpaths in `"exports"` to all be entry points in the package. That was the thinking behind the previous heading "Package Entry Points."

Reply:
The change here is that previously we were not treating "exports" as the primary way of setting the main entry point to the package.

But because of the added benefits, we really do want to encourage users to define "exports" for the various benefits of doing that.

By explicitly saying "exports" is how you define package entries for ESM we are jumping to that advice, and then just noting that "main" is still supported. It is strong I know, but I think it can be useful to do.

There is also the argument we shouldn't be dictating to users how to do things though too, and I'm open to discussing in more detail if we really do want to be encouraging exports-first for the main.

Reply:
I agree we should be pretty directive that users should use `"exports"`. What about keeping the previous heading Package Entry Points, then covering all the `"exports"` stuff, and then the last subsection is about `"main"` and how it also applies for `import` (i.e. the previous `"main"` subsection)? Just so that it's not undocumented that `"main"` is used in ESM.

Reply:
Ok I've renamed the section to package entry points, then included another paragraph that more generally explains the module format situation and points to the dual mode section. I quite like explaining it as a problem for all entry points and not just the main. Further feedback welcome.